### PR TITLE
COCOS-9 - Dockerize manager

### LINKED ISSUE #9: Deploy Manager in Docker
Explore how Manager can be deployed in Docker. It needs to talk to the underlying host, hopefully via `libvirt` rather than Qemu command line.

### PR DESCRIPTION
**Pull request by SammyOina**
Dockerize manager
Resolves #9 

Notes:
- When testing, agent needs to be running and set AGENT_GRPC_URL variable in .env
- Also during tests edit docker-compose to the correct volume paths for OVMF_VARS.fd and OVMF_CODE.fd on your system.

### CODE REVIEW

File: cmd/cli/main.go [hunk 1]
```diff
@@ -41,6 +42,7 @@ func main() {
 	if err := env.Parse(&agentGRPCConfig, env.Options{Prefix: envPrefixAgentGRPC}); err != nil {
 		logger.Fatal(fmt.Sprintf("failed to load %s gRPC client configuration : %s", svcName, err))
 	}
+	agentGRPCConfig.URL = defURL
```
Comment (rodneyosodo) [inline cmd/cli/main.go:45]: This shouldn't be the case. You are effectively writing the deafultURL to a fixed values after loading the env configs. Have a look at https://github.com/caarlos0/env?tab=readme-ov-file#defaults-from-code. You can specify the default values when loading env vars

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: cmd/cli/main.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ import (
 const (
 	svcName            = "cli"
 	envPrefixAgentGRPC = "AGENT_GRPC_"
+	defURL             = "localhost:7002"
```
Comment (rodneyosodo) [inline cmd/cli/main.go:23]: This is now unused

File: manager/README.md [hunk 3]
```diff
@@ -21,8 +21,9 @@ The service is configured using the environment variables from the following tab
 | AGENT_GRPC_TIMEOUT       | Agent service gRPC timeout                               | 1s                                |
 | AGENT_GRPC_CA_CERTS      | Agent service gRPC CA certificates                       |                                   |
 | AGENT_GRPC_CLIENT_TLS    | Agent service gRPC client TLS                            | false                             |
-| MANAGER_JAEGER_URL       | Jaeger server URL                                        | http://localhost:14268/api/traces |
+| COCOS_JAEGER_URL         | Jaeger server URL                                        | http://localhost:14268/api/traces |
 | MANAGER_INSTANCE_ID      | Manager service instance ID                              |                                   |
+| COCOS_MESSAGE_BROKER_URL | message broker url                                       | http://localhost:4222             |
```
Comment (rodneyosodo) [inline manager/README.md:26]:
```suggestion
| COCOS_MESSAGE_BROKER_URL | message broker url                                       | nats://localhost:4222             |
```

File: agent/agent.pb.go [hunk 4]
```diff
@@ -4,7 +4,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
 // 	protoc-gen-go v1.31.0
-// 	protoc        v4.24.4
+// 	protoc        v4.25.0
```
Comment (rodneyosodo) [inline agent/agent.pb.go:7]: Also update version in CI

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: docker/brokers/docker-compose.yml [hunk 5]
```diff
@@ -1,22 +1,15 @@
...
       service: broker
     container_name: cocos-broker
     restart: on-failure
     networks:
       - cocos-base-net
     volumes:
       - cocos-broker-volume:/data
-    profiles:
-      - rabbitmq
-      - nats
```
Comment (rodneyosodo) [inline docker/brokers/docker-compose.yml:22]: Add new line

File: docker/brokers/docker-compose.yml [hunk 6]
```diff
@@ -1,22 +1,15 @@
 # This file configures COCOS brokers. 
 # MESSAGE_BROKER can be either 'nats' or 'rabbitmq'.
 #
-# Each broker has a unique profile for configuration. The available profiles are:
-# - nats: Uses 'nats' as MESSAGE_BROKER.
-# - rabbitmq: Uses 'rabbitmq' as MESSAGE_BROKER.
-#
 
 services:
   broker:
```
Comment (rodneyosodo) [inline docker/brokers/docker-compose.yml:6]: Is this being imported somewhere? I am not seeing it

File: docker/.env [hunk 7]
```diff
@@ -16,8 +16,8 @@ COCOS_RABBITMQ_VHOST=/
 COCOS_RABBITMQ_URL=amqp://${COCOS_RABBITMQ_USER}:${COCOS_RABBITMQ_PASS}@broker:${COCOS_RABBITMQ_PORT}${COCOS_RABBITMQ_VHOST}
 
 ## Message Broker
-COCOS_MESSAGE_BROKER_TYPE=nats
-COCOS_MESSAGE_BROKER_URL=${COCOS_NATS_URL}
+COCOS_MESSAGE_BROKER_TYPE=rabbitmq
```
Comment (rodneyosodo) [inline docker/.env:19]: It should be Nats

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: docker/brokers/docker-compose.yml (hunk 6 above)
Comment (SammyOina) [reply docker/brokers/docker-compose.yml:6]: https://github.com/ultravioletrs/cocos-ai/blob/7b31e2181828e21c6d2a00eb20dc7c6b35dd4f2a/docker/docker-compose.yml#L12

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**